python_modules/libraries/dagster-polars/dagster_polars/io_managers/parquet.py

```diff
@@ -0,0 +1,201 @@
+import json
+from typing import TYPE_CHECKING, Any, Optional, Union
+
+import polars as pl
+import pyarrow as pa
+import pyarrow.dataset
+import pyarrow.parquet
+from dagster import InputContext, OutputContext
+from packaging.version import Version
+from pyarrow import Table
+from upath import UPath
+
+from dagster_polars.constants import DAGSTER_POLARS_STORAGE_METADATA_KEY
+from dagster_polars.io_managers.base import BasePolarsUPathIOManager
+from dagster_polars.types import LazyFrameWithMetadata, StorageMetadata
+
+if TYPE_CHECKING:
+    import fsspec
+
+
+def get_pyarrow_dataset(path: UPath, context: InputContext) -> pyarrow.dataset.Dataset:
+    assert context.metadata is not None
+
+    fs: Union[fsspec.AbstractFileSystem, None] = None
+
+    try:
+        fs = path.fs
+    except AttributeError:
+        pass
+
+    ds = pyarrow.dataset.dataset(
+        str(path),
+        filesystem=fs,
+        format=context.metadata.get("format", "parquet"),
+        partitioning=context.metadata.get("partitioning"),
+        partition_base_dir=context.metadata.get("partition_base_dir"),
+        exclude_invalid_files=context.metadata.get("exclude_invalid_files", True),
+        ignore_prefixes=context.metadata.get("ignore_prefixes", [".", "_"]),
+    )
+
+    return ds
+
+
+def scan_parquet_legacy(path: UPath, context: InputContext) -> pl.LazyFrame:
+    """Scan a parquet file and return a lazy frame (uses pyarrow).
```

Comment:
Is legacy the right word here? Is the only differencep yarrow vs polars?

Reply:
I am calling it "legacy" because prior to ` polars 0.19.4` `pyarrow` was always use to scan parquets from remote filesystems. 

Once `0.19.4` was released I switched to native `pl.scan_parquet`, but kept the legacy scanning method for backwards compatibility for users with older polars versions.